sdk/aws-autoloader.php

```diff
@@ -2826,8 +2826,13 @@
     }
 }, true);
 
-require __DIR__ . '/Aws/functions.php';
-require __DIR__ . '/GuzzleHttp/functions_include.php';
+if (!file_exists(__DIR__ . '/../../../../libraries/optional/aws/aws-sdk-php/src/functions.php') && !function_exists('Aws\constantly'))  {
```

Comment:
Replace the morse code with `$CFG->libraries`

Reply:
what stable branches is CFG->libraries supported on?

Reply:
It's in Totara 13+, so maybe just `if (!isset($CFG->libraries) && [...]` would be more appropriate

Reply:
If so make sure the comment is clear saying exactly  why its doing this and on what versions